pkg/catalogv2/git/download.go

```diff
@@ -1,124 +1,89 @@
 package git
 
 import (
-	"crypto/sha256"
-	"encoding/hex"
-	"encoding/pem"
 	"fmt"
-	"net/url"
-	"os"
-	"path/filepath"
-	"regexp"
-	"strings"
-
-	"github.com/rancher/rancher/pkg/settings"
-	corev1 "k8s.io/api/core/v1"
 )
 
-const (
-	stateDir  = "management-state/git-repo"
-	staticDir = "/var/lib/rancher-data/local-catalogs/v2"
-	localDir  = "../rancher-data/local-catalogs/v2" // identical to helm.InternalCatalog
-)
-
-func gitDir(namespace, name, gitURL string) string {
-	staticDir := filepath.Join(staticDir, namespace, name, hash(gitURL))
-	if s, err := os.Stat(staticDir); err == nil && s.IsDir() {
-		return staticDir
+// Ensure will check if repo is cloned, if not the method will clone and reset to the latest commit.
+// If reseting to the latest commit is not possible it will fetch and try to reset again
+func (r *repository) Ensure(branch string) error {
+	err := r.cloneOrOpen("")
+	if err != nil {
+		return err
```

Comment:
We are wrapping errors in some functions and not wrapping in others, we should keep it consistent

Reply:
All errors are wrapped at **git.go** our go-git implementation. 
All errors at **download.go** are received from **git.go** therefore they do not wrap any errors.
It is consistent with what I mentioned.
However, I agree with you that would be good to have a 2nd level of wrapping at **download.go** to be more easier to locate where the error happened. I was following the same behavior as before the implementation trying to minimize my footprint.  
I will make your suggested changes

Reply:
Thanks, that will make it easier to debug if needed to